test/controllers/setting_controller_test.exs

```diff
@@ -0,0 +1,122 @@
+defmodule CoursePlanner.SettingControllerTest do
+  use CoursePlanner.ConnCase
+
+  alias CoursePlanner.SystemVariable
+  import CoursePlanner.Factory
+
+  defp login_as(user_type) do
+    user = insert(user_type)
+
+    Phoenix.ConnTest.build_conn()
+    |> assign(:current_user, user)
+  end
+
+  test "shows chosen resource only for coordinator", %{conn: _conn} do
+    coordinator_conn = login_as(:coordinator)
+
+    conn = get coordinator_conn, setting_path(coordinator_conn, :show)
+    assert html_response(conn, 200) =~ "Show settings"
+  end
+
+  test "does not show chosen resource for non coordinator users", %{conn: _conn} do
```

Comment:
Could you check something like the offered_course_controller_test.exs? 

These tests contain too much information, and in the other hand if they fail there is not enough information, so it is hard to understand why they failed